src/lib.rs

```diff
@@ -870,58 +987,16 @@ impl List {
         }
         n -= self.notified;
 
-        while n > 0 {
+        while n > 0 && self.notify_internal(false, ()) {
```

Comment:
These while loops are probably a bit too cute. Happy to rewrite them as this if you think mutating state in the conditional is too confusing:

```rust
while n > 0 {
    n -= 1;
    if self.notify_internal(false, ()) {
        break;
    }
}
```